scripting/include/srccoop/manager.inc

```diff
@@ -170,123 +234,56 @@ methodmap CoopManager
 			else if (g_pLevelLump.m_iIntroType == INTRO_FREEZE)
 			{
 				pPlayer.SetFlags(pPlayer.GetFlags() | FL_ATCONTROLS);
-				Client_ScreenFade(pPlayer.GetEntIndex(), Conf.FROZEN_FADE_DUR_IN, FFADE_OUT|FFADE_STAYOUT, _,
+				Client_ScreenFade(pPlayer.GetEntIndex(), Conf.FROZEN_FADE_DUR_IN, FFADE_OUT | FFADE_STAYOUT, _,
 					Conf.FROZEN_FADE_COLOR[0], Conf.FROZEN_FADE_COLOR[1], Conf.FROZEN_FADE_COLOR[2], Conf.FROZEN_FADE_COLOR[3]);
 			}
-			CoopManager.CheckStart();
-		}
-	}
 
-	public static void CheckStart()
-	{
-		if (!data.m_bStarted && data.m_bPlayerSpawned)
-		{
-			// are we allowed to start or waiting for players?
-			if (CoopManager.IsAllowedToStart())
+			if ((g_pLevelLump.m_iIntroType == INTRO_NONE && !data.m_bDoesMapHaveDelayedOutput))
 			{
-				// is initial timer already running?
-				if (CoopManager.IsInCountDownPeriod())
+				if (data.m_hStartTimer)
 				{
-					// can start, but do a 3 sec readyup timer as we cancel the initial player wait timer
-					if (!data.m_bGetReadyTimer)
-					{
-						delete data.m_pStartTimer;
-						CoopManager.StartCountdown(true);
-					}
+					delete data.m_hStartTimer;
+					data.m_hStartTimer = null;
 				}
-				else
+				data.m_iSecondsToStart = 0;
+				Timer_DeincrementUntilMapStart(null);
+			}
+			else if (((GetGameTime() - data.m_flMapStartTime) <= g_flWaitPeriodSkipGraceTime) && (data.m_iSpawnedPlayerCountLastMap - data.m_iSpawnedPlayerCount) == 0)
+			{
+				if (!data.m_hStartTimer || data.m_iSecondsToStart > 3)
 				{
-					// we can just start if no previous timer is running
-					CoopManager.Start();
+					if (data.m_hStartTimer)
+					{
+						delete data.m_hStartTimer;
```

Comment:
All 5 lines starting with `if` can be just `delete data.m_hStartTimer;`
`delete` checks for null, closes, and then sets null

Reply:
Fixed two instances.